cadc-data-ops-fits/src/main/java/org/opencadc/fits/slice/Slices.java

```diff
@@ -0,0 +1,306 @@
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.fits.slice;
+
+import org.apache.log4j.Logger;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+
+/**
+ * Class to represent a full slice request.
+ */
+public final class Slices {
+    private static final Logger LOGGER = Logger.getLogger(Slices.class);
+    private static final String PIXEL_RANGE_PATTERN_STRING = "\\[([\\d*]+:?\\d*,?\\s*)*\\]";
+    private static final String ENTIRE_RANGE_PATTERN_STRING =
+            "(\\s*\\[[\\w]*,?\\s*\\d*\\])?" + PIXEL_RANGE_PATTERN_STRING;
+
+    private final ExtensionSliceValue[] extensionSliceValues;
+
+    public static Slices fromString(final String value) {
+        System.out.println("Matching on " + value);
+        final Pattern entireRangePattern = Pattern.compile(ENTIRE_RANGE_PATTERN_STRING);
+        final List<ExtensionSliceValue> extensionSliceValues = matchAndCollect(entireRangePattern, value);
+
+        if (extensionSliceValues.isEmpty()) {
+            throw new IllegalArgumentException("No parsable values in " + value);
+        } else {
+            return new Slices(extensionSliceValues.toArray(new ExtensionSliceValue[0]));
+        }
+    }
+
+    private static List<ExtensionSliceValue> matchAndCollect(final Pattern pattern, final String value) {
+        final Matcher matcher = pattern.matcher(value);
+        final List<ExtensionSliceValue> extensionSliceValues = new ArrayList<>();
+
+        while (matcher.find()) {
+            final String nextMatch = matcher.group();
+            System.out.println("Next match is " + nextMatch);
+            extensionSliceValues.add(ExtensionSliceValue.fromString(nextMatch));
+        }
+
+        return extensionSliceValues;
+    }
+
+    Slices(final ExtensionSliceValue[] extensionSliceValues) {
+        this.extensionSliceValues = extensionSliceValues;
+    }
+
+    public boolean hasRanges() {
+        return this.extensionSliceValues != null && this.extensionSliceValues.length > 0;
+    }
+
+    public ExtensionSliceValue[] getExtensionSliceValues() {
+        return this.extensionSliceValues;
+    }
+
+
+    /**
+     * Represents a single extension and set of pixel ranges.
+     */
+    public static final class ExtensionSliceValue {
+
+        private final static String ALL_DATA = "*";
+        private final static String SEPARATOR = ",";
+        private final static String PIXEL_VALUE_DELIMITER = ":";
+        private static final Pattern VALID_SINGLE_RANGE = Pattern.compile("[\\d*\\*]+:?\\d*");
```

Comment:
The pattern value here is a subset of PIXEL_RANGE_PATTERN_STRING so this should be defined above that and used when creating PIXEL_RANGE_PATTERN_STRING - because they do have to agree on this.